mmdet/core/hook/memory_profiler_hook.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from mmcv.runner.hooks import HOOKS, Hook
+
+try:
+    import psutil
+    from memory_profiler import memory_usage
+except ImportError:
+    psutil = None
+    memory_usage = None
+
+
+@HOOKS.register_module()
+class MemoryProfilerHook(Hook):
+    """Memory profiler hook recording memory information: virtual memory, swap
+    memory and memory of current process.
+
+    Args:
+        interval (int): Checking interval (every k iterations).
+            Default: 50.
+    """
+
+    def __init__(self, interval=50):
+        if psutil is None and memory_usage is None:
+            raise RuntimeError('Please install psutil and memory_profiler.')
+        elif psutil is None:
+            raise RuntimeError('Please install psutil.')
+        elif memory_usage is None:
+            raise RuntimeError('Please install memory_profiler.')
+
+        self.interval = interval
+
+    def after_iter(self, runner):
```

Comment:
Is the validation phase also required?

Reply:
Yes, there are differences between train phase and val phase, such as hooks called in difference phases.